amazon_cloud_code_generator/api_specifications/AWS::Backup::BackupVault.json

```diff
@@ -0,0 +1 @@
+{"typeName": "AWS::Backup::BackupVault", "description": "Resource Type definition for AWS::Backup::BackupVault", "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-rpdk.git", "definitions": {"NotificationObjectType": {"type": "object", "additionalProperties": false, "properties": {"BackupVaultEvents": {"type": "array", "insertionOrder": false, "uniqueItems": false, "items": {"type": "string"}}, "SNSTopicArn": {"type": "string"}}, "required": ["SNSTopicArn", "BackupVaultEvents"]}, "LockConfigurationType": {"type": "object", "additionalProperties": false, "properties": {"MinRetentionDays": {"type": "number"}, "MaxRetentionDays": {"type": "number"}, "ChangeableForDays": {"type": "number"}}, "required": ["MinRetentionDays"]}, "BackupVaultNamePattern": {"type": "string", "pattern": "^[a-zA-Z0-9\\-\\_]{2,50}$"}}, "properties": {"AccessPolicy": {"type": ["object", "string"]}, "BackupVaultName": {"$ref": "#/definitions/BackupVaultNamePattern"}, "BackupVaultTags": {"type": "object", "additionalProperties": false, "patternProperties": {"^.{1,128}$": {"type": "string"}}}, "EncryptionKeyArn": {"type": "string"}, "Notifications": {"$ref": "#/definitions/NotificationObjectType"}, "LockConfiguration": {"$ref": "#/definitions/LockConfigurationType"}, "BackupVaultArn": {"type": "string"}}, "additionalProperties": false, "taggable": true, "required": ["backup_vault_name"], "createOnlyProperties": ["backup_vault_name", "encryption_key_arn"], "readOnlyProperties": ["backup_vault_arn"], "primaryIdentifier": ["backup_vault_name"], "handlers": {"create": {"permissions": ["backup:CreateBackupVault", "backup:PutBackupVaultAccessPolicy", "backup:PutBackupVaultNotifications", "backup:PutBackupVaultLockConfiguration", "backup-storage:Mount", "backup-storage:MountCapsule", "kms:CreateGrant", "kms:GenerateDataKey", "kms:Decrypt", "kms:RetireGrant", "kms:DescribeKey"]}, "read": {"permissions": ["backup:DescribeBackupVault", "backup:GetBackupVaultNotifications", "backup:GetBackupVaultAccessPolicy", "backup:ListTags"]}, "update": {"permissions": ["backup:DescribeBackupVault", "backup:DeleteBackupVaultAccessPolicy", "backup:DeleteBackupVaultNotifications", "backup:DeleteBackupVaultLockConfiguration", "backup:ListTags", "backup:TagResource", "backup:UntagResource", "backup:PutBackupVaultAccessPolicy", "backup:PutBackupVaultNotifications", "backup:PutBackupVaultLockConfiguration"]}, "delete": {"permissions": ["backup:DeleteBackupVault"]}, "list": {"permissions": ["backup:ListBackupVaults"]}}}
```

Comment:
Why host schemas in a folder? AWS is constantly updating schemas, and using CloudFormation to extract the schema on the fly also prevents the use of outdated schemas. It seems to me that manually downloading the schemas and putting them in a dedicated folder is of no benefit. I'm not sure, but we probably don't want that to be the case. I think it would be easier for users who want to use the generator for personal use in general to specify the name of the resource they want to use and generate the corresponding module form instead of doing all that.

Not sure what @jillr thinks about.

Reply:
This commit doesn't change the way the modules are generated. It's just split the process in two tasks. From a testing perspective, this we will allow us to control the input we provide to `refresh_modules`, we will also be able to validate the output. This will also give us the ability to run the tests offline. Finally, in the future, we may want to prepare a minor update for a released stable version, in this case, we may also need the schema that were used initially.